src/components/views/rooms/AudioRecorder.js

```diff
@@ -0,0 +1,82 @@
+//@flow
+
+import React from 'react';
+
+import sdk from '../../../index';
+import Modal from '../../../Modal';
+import ErrorDialog from '../dialogs/ErrorDialog';
+import MediaStreamRecorder from 'msr';
+
+export default class AudioRecorder extends React.Component {
+  static propTypes = {
+    sendAudioBlob: React.PropTypes.func.isRequired,
+  };
+
+  mediaStream: ?MediaStream = null;
+  recorder: ?MediaStreamRecorder = null;
+  blobs: Array<Blob> = [];
+
+  constructor(props: Object) {
+    super(props);
+
+    this.state = {
+      recording: false,
+    };
+  }
+
+  onRecordClicked = async () => {
+    if(this.state.recording && this.mediaStream) {
+      this.recorder.stop();
+      window.ConcatenateBlobs(this.blobs, this.blobs[0].type, (blob) => {
+        console.log(blob);
+        this.props.sendAudioBlob(blob);
+        this.blobs = [];
+        this.recorder = null;
+        this.mediaStream.getTracks().forEach(track => track.stop());
+        this.mediaStream = null;
+      });
+    } else {
+      try {
+        const mediaStream = await navigator.mediaDevices.getUserMedia({
+          audio: true,
+        });
+        this.onUserMediaSuccess(mediaStream);
+      } catch (e) {
+        this.onUserMediaFailure(e);
+      }
+    }
+  };
+
+  onUserMediaFailure = () => {
+    Modal.createDialog(ErrorDialog, {
+      description: "Failed to start recording.",
+    });
+  };
+
+  onUserMediaSuccess = (mediaStream: MediaStream) => {
+    this.mediaStream = mediaStream;
+    this.recorder = new MediaStreamRecorder(this.mediaStream);
+    this.recorder.recorderType = class extends MediaStreamRecorder.StereoAudioRecorder {
```

Comment:
If this were a named class you could call it MonoAudioRecorder which might make it more obvious what's going on.